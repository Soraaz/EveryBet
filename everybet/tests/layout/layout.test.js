import React from "react";
import {TouchableWithoutFeedback} from 'react-native';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import Layout from "../../src/layout/layout";
import User from "../../src/classes/user";

describe("layout", () => {
	describe("rendering", () => {
		it("Should render correctly", () => {
			const tree = renderer
				.create(<Layout />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly with a connected user", () => {
			User.coins = 150;
			const tree = renderer
				.create(<Layout />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly with a connected user and a custom avatar", () => {
			User.avatar = "avatar";
			const tree = renderer
				.create(<Layout />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {
		it("Should navigate to Home pressing the logo", () => {
			const navigation = {navigate: jest.fn(x => x)};
			const wrapper = shallow(<Layout navigation={navigation} />);
			expect(wrapper.find(TouchableWithoutFeedback)).to.have.lengthOf(2);
			expect(wrapper.find(TouchableWithoutFeedback).first().props().onPress()).to.be.equal('Home');
		});

		it("Should navigation to Profile when the user's icon", () => {
			const navigation = {navigate: jest.fn(x => x)};
			const wrapper = shallow(<Layout navigation={navigation} />);
			expect(wrapper.find(TouchableWithoutFeedback).at(1).props().onPress()).to.be.equal('Profile');
		});
	});
});
