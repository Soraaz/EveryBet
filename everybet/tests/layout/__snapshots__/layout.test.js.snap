// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout rendering Should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#37597F",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../assets/images/everybet_logo.png",
      }
    }
    style={
      Object {
        "height": 39.25,
        "marginLeft": 25,
        "width": 138,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "flex-start",
        "marginLeft": "auto",
        "width": 90,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/betcoin_icon.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "System",
          "fontSize": 10,
          "marginLeft": "auto",
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    >
      0
    </Text>
  </View>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 15,
        "width": 40,
      }
    }
  />
</View>
`;

exports[`layout rendering Should render correctly with a connected user 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#37597F",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../assets/images/everybet_logo.png",
      }
    }
    style={
      Object {
        "height": 39.25,
        "marginLeft": 25,
        "width": 138,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "flex-start",
        "marginLeft": "auto",
        "width": 90,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/betcoin_icon.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "System",
          "fontSize": 10,
          "marginLeft": "auto",
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    >
      150
    </Text>
  </View>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../assets/images/user_placeholder.png",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 15,
        "width": 40,
      }
    }
  />
</View>
`;

exports[`layout rendering Should render correctly with a connected user and a custom avatar 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#37597F",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../assets/images/everybet_logo.png",
      }
    }
    style={
      Object {
        "height": 39.25,
        "marginLeft": 25,
        "width": 138,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "flex-start",
        "marginLeft": "auto",
        "width": 90,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/betcoin_icon.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "System",
          "fontSize": 10,
          "marginLeft": "auto",
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    >
      150
    </Text>
  </View>
  <Image
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "method": "GET",
        "uri": "avatar",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginHorizontal": 15,
        "width": 40,
      }
    }
  />
</View>
`;
